system/lib/libc/musl/src/locale/newlocale.c

```diff
@@ -7,7 +7,18 @@ locale_t newlocale(int mask, const char *name, locale_t base)
 {
 	if (*name && strcmp(name, "C") && strcmp(name, "POSIX"))
 		return 0;
-	if (!base) base = calloc(1, sizeof *base);
+	if (!base) {
```

Comment:
I wonder if this is a bit nasty. Do we know if there will only be at most one `newlocale` call at startup so we get the full benefits of this optimization?